### DISCUSSION TITLE
Migrating one-to-many to many-to-many

### DISCUSSION THREAD

**Discussion by rohinb2** [sqlalchemy/sqlalchemy#6792] [Usage Questions]
Hi, I currently have some tables that have a one-to-many relationship, e.g:
```
A:
    name

B:
    name
    A.id (fk)
```
However, I want to migrate this to have a junction table with a many to many relationship. There are plenty of docs for writing the SQL alchemy for a many to many relationship, but few docs that actually show the proper way to write the migration such that I can migrate the data over. Can anyone provide guidance on this?

**Comment by CaselIT** [comment]
You may want to check alembic for this, but it does not handle automatically the migration from one to many to many to many.

I think the steps would be:
- add the relation table
- insert in the relation table all the current one to many relations
- remove the old fk on the many side table of old one to many relation
